hi-ml/src/health_ml/runner.py

```diff
@@ -325,15 +326,29 @@ def run_in_situ(self, azure_run_info: AzureRunInfo) -> None:
             assert azure_run_info.run is not None
             azure_run_info.run.set_tags(self.additional_run_tags(sys.argv[1:]))
 
-        # Set environment variables for multi-node training if needed. This function will terminate early
-        # if it detects that it is not in a multi-node environment.
-        if self.experiment_config.num_nodes > 1:
-            set_environment_variables_for_multi_node()
-        self.ml_runner = MLRunner(
-            experiment_config=self.experiment_config, container=self.lightning_container, project_root=self.project_root
-        )
-        self.ml_runner.setup(azure_run_info)
-        self.ml_runner.run()
+        if self.experiment_config.mode == RunnerMode.TRAIN:
+            # Set environment variables for multi-node training if needed. This function will terminate early
+            # if it detects that it is not in a multi-node environment.
+            if self.experiment_config.num_nodes > 1:
+                set_environment_variables_for_multi_node()
+            self.ml_runner = TrainingRunner(
+                experiment_config=self.experiment_config,
+                container=self.lightning_container,
+                project_root=self.project_root,
+            )
+            self.ml_runner.validate()
+            self.ml_runner.run_and_cleanup(azure_run_info)
+        elif self.experiment_config.mode == RunnerMode.EVAL:
+            self.eval_runner = EvalRunner(
```

Comment:
use same attribute self.ml_runner for both (see comment above about typing) and call 
self.ml_runner.validate()
self.ml_runner.run_and_cleanup(azure_run_info)
outside the if else statement (TODO for DRY)